tools/vargo/src/util.rs

```diff
@@ -66,3 +66,65 @@ pub fn mtime_recursive(path: &Path) -> Result<FileTime, String> {
         .unwrap_or_else(|| FileTime::from_last_modification_time(&meta));
     Ok(max_meta)
 }
+
+pub fn commit_info() -> Result<String, std::io::Error> {
+    // assumes the verus executable gets the .git file for verus repo
+    let mut exe_dir = std::env::current_exe().unwrap();
+    exe_dir.pop();
+
+    let sha = std::process::Command::new("git")
+        .current_dir(&exe_dir)
+        .args(&["rev-parse", "--short", "HEAD"])
+        .stdout(std::process::Stdio::piped())
+        .spawn()?;
+
+    let mut sha_msg = sha
+        .wait_with_output()
+        .expect("failed to execute git rev-parse HEAD")
+        .stdout;
+    sha_msg.pop();
+
+    let date_info = std::process::Command::new("git")
+        .current_dir(&exe_dir)
+        .args(&["show", "-s", "--format=%ci", "HEAD"])
+        .stdout(std::process::Stdio::piped())
+        .spawn()?;
+
+    let date_msg = date_info
+        .wait_with_output()
+        .expect("failed to execute git show -s --format=%ci HEAD")
+        .stdout;
+
+    let year = String::from_utf8(date_msg[0..4].to_vec()).unwrap();
+    let month = String::from_utf8(date_msg[5..7].to_vec()).unwrap();
+    let day = String::from_utf8(date_msg[8..10].to_vec()).unwrap();
+
+    let child = std::process::Command::new("git")
+        .current_dir(&exe_dir)
+        .args(&["diff", "--exit-code", "HEAD"])
+        .stdout(std::process::Stdio::null())
+        .spawn()?;
+
+    let status = child
+        .wait_with_output()
+        .expect("failed to execute git diff --exit-code HEAD")
+        .status;
+
+    if status.success() {
+        Ok(format!(
+            "0.{}.{}.{}-{}",
+            year,
+            month,
+            day,
+            String::from_utf8(sha_msg).unwrap()
+        ))
+    } else {
+        Ok(format!(
+            "0.{}.{}.{}-{} (dirty)",
+            year,
+            month,
+            day,
+            String::from_utf8(sha_msg).unwrap()
+        ))
+    }
```

Comment:
A somewhat cleaner way to handle this:
```suggestion
    let dirty = if status.success() { "" } else { ".dirty" };
    Ok(format!("0.{year}.{month}.{day}{dirty}"))
```